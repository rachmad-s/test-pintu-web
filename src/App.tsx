import { QueryClient, QueryClientProvider } from "react-query";
import CurrencyContext from "./contexts/CurrencyContext";

import Market from "./pages/Market";
import useTokenList from "./hooks/TokenList";

const queryClient = new QueryClient();

function InnerApp() {
  const { isLoading, error, data, filter, setFilter } = useTokenList();
  return (
    <CurrencyContext.Provider
      value={{
        isLoading,
        error,
        data,
        filter,
        setFilter,
      }}
    >
      <Market />
    </CurrencyContext.Provider>
  );
}

function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <InnerApp />
    </QueryClientProvider>
  );
}

export default App;
