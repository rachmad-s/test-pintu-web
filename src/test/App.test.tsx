import "@testing-library/jest-dom";
import { render, cleanup, screen } from "@testing-library/react";
import App from "../App";
import { getCurrenciesRes, API, getPriceChangesRes } from "./mocks/response";
afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});

it("should loading the table", () => {
  render(<App />);

  const table = screen.getByTestId("table-currency");
  expect(table).toHaveClass("loading-skeleton");
});

it("should load data from API", () => {
  render(<App />);

  jest
    .spyOn(API, "getCurrencies")
    .mockImplementation(() => Promise.resolve(getCurrenciesRes));

  jest
    .spyOn(API, "getPriceChanges")
    .mockImplementation(() => Promise.resolve(getPriceChangesRes));
});
