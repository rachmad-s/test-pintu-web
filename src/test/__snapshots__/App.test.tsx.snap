// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container pt-10 pb-3"
  >
    <div
      class="lg:flex justify-between items-start mb-10 px-4 sm:px-0"
    >
      <h1
        class="text-3xl font-bold lg:mb-0 mb-10"
      >
        Harga Crypto dalam Rupiah Hari ini
      </h1>
      <div
        class="flex mt-0"
      >
        <span
          class="p-2 px-3 border-t border-b border-l border-gray bg-gray/60 rounded-tl-lg rounded-bl-lg"
        >
          <svg
            class="text-gray-dark mt-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
        </span>
        <input
          class="bg-transparent w-80 border-t border-b border-r border-gray bg-gray/60 rounded-tr-lg rounded-br-lg p-2 focus:outline-none"
          placeholder="Cari aset di Pintu"
          type="text"
        />
      </div>
    </div>
    <div
      class="flex px-4 sm:px-0 nowrap overflow-x-auto gap-3 pb-3"
    >
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3v4M3 5h4M6 17v4m-2-2h4m5-16l2.286 6.857L21 12l-5.714 2.143L13 21l-2.286-6.857L5 12l5.714-2.143L13 3z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        Terbaru
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M6.5 10h-2v7h2v-7zm6 0h-2v7h2v-7zm8.5 9H2v2h19v-2zm-2.5-9h-2v7h2v-7zm-7-6.74L16.71 6H6.29l5.21-2.74m0-2.26L2 6v2h19V6l-9.5-5z"
          />
        </svg>
        DeFi
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M369.3 146H142.7C81.5 146 32 191.5 32 255.4c0 64 49.5 110.6 110.7 110.6h226.5c61.2 0 110.7-46.6 110.7-110.6.1-63.9-49.4-109.4-110.6-109.4zM200 266.7c0 2.7-2.4 5.3-5.2 5.3H160v35.1c0 2.8-3.1 4.9-5.8 4.9h-21.4c-2.6 0-4.8-1.9-4.8-4.5V272H92.9c-2.8 0-4.9-3.1-4.9-5.8v-21.4c0-2.6 1.9-4.8 4.5-4.8H128v-34.8c0-2.8 1.9-5.2 4.6-5.2h22.1c2.7 0 5.3 2.4 5.3 5.2V240h34.8c2.8 0 5.2 1.9 5.2 4.6v22.1zm119.8 8.8c-10.7 0-19.5-8.6-19.5-19.2s8.7-19.2 19.5-19.2 19.5 8.6 19.5 19.2-8.8 19.2-19.5 19.2zm42.5 41.8c-10.7 0-19.5-8.5-19.5-19.1 0-10.6 8.7-19.2 19.5-19.2s19.5 8.5 19.5 19.2c0 10.6-8.7 19.1-19.5 19.1zm0-83.7c-10.7 0-19.5-8.6-19.5-19.1 0-10.6 8.7-19.2 19.5-19.2s19.5 8.6 19.5 19.2c0 10.5-8.7 19.1-19.5 19.1zm42.6 41.9c-10.7 0-19.4-8.6-19.4-19.2s8.7-19.2 19.4-19.2 19.5 8.6 19.5 19.2c-.1 10.6-8.8 19.2-19.5 19.2z"
          />
        </svg>
        NFT/Gaming
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 2a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v12h.5a.5.5 0 0 1 0 1H.5a.5.5 0 0 1 0-1H1v-3a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v3h1V7a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v7h1V2zm1 12h2V2h-2v12zm-3 0V7H7v7h2zm-5 0v-3H2v3h2z"
          />
        </svg>
        CEX
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 11.5a.5.5 0 0 0 .5.5h11.793l-3.147 3.146a.5.5 0 0 0 .708.708l4-4a.5.5 0 0 0 0-.708l-4-4a.5.5 0 0 0-.708.708L13.293 11H1.5a.5.5 0 0 0-.5.5zm14-7a.5.5 0 0 1-.5.5H2.707l3.147 3.146a.5.5 0 1 1-.708.708l-4-4a.5.5 0 0 1 0-.708l4-4a.5.5 0 1 1 .708.708L2.707 4H14.5a.5.5 0 0 1 .5.5z"
            fill-rule="evenodd"
          />
        </svg>
        DEX
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.235 1.559a.5.5 0 0 0-.47 0l-7.5 4a.5.5 0 0 0 0 .882L3.188 8 .264 9.559a.5.5 0 0 0 0 .882l7.5 4a.5.5 0 0 0 .47 0l7.5-4a.5.5 0 0 0 0-.882L12.813 8l2.922-1.559a.5.5 0 0 0 0-.882l-7.5-4zM8 9.433 1.562 6 8 2.567 14.438 6 8 9.433z"
          />
        </svg>
        Layer-1
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 7v10c0 2.21 3.582 4 8 4s8-1.79 8-4V7M4 7c0 2.21 3.582 4 8 4s8-1.79 8-4M4 7c0-2.21 3.582-4 8-4s8 1.79 8 4m0 5c0 2.21-3.582 4-8 4s-8-1.79-8-4"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        Infrastructure
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-3.5-6H14a.5.5 0 1 0 0-1h-4a2.5 2.5 0 1 1 0-5h1V6h2v2h2.5v2H10a.5.5 0 1 0 0 1h4a2.5 2.5 0 1 1 0 5h-1v2h-2v-2H8.5v-2z"
            />
          </g>
        </svg>
        Lending
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.235 1.559a.5.5 0 0 0-.47 0l-7.5 4a.5.5 0 0 0 0 .882L3.188 8 .264 9.559a.5.5 0 0 0 0 .882l7.5 4a.5.5 0 0 0 .47 0l7.5-4a.5.5 0 0 0 0-.882L12.813 8l2.922-1.559a.5.5 0 0 0 0-.882l-7.5-4zM8 9.433 1.562 6 8 2.567 14.438 6 8 9.433z"
          />
        </svg>
        Layer-2
      </button>
      <button
        class="flex bg-blue text-blue text-sm items-center gap-1 rounded-lg p-2 px-4 whitespace-nowrap"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.63 7L13 3h1V2H9V1H8v1H3v1h1L2.38 7H2v1h.15c.156.498.473.93.9 1.23a2.47 2.47 0 0 0 2.9 0A2.44 2.44 0 0 0 6.86 8H7V7h-.45L4.88 3H8v8H6l-.39.18-2 2.51.39.81h9l.39-.81-2-2.51L11 11H9V3h3.13l-1.67 4H10v1h.15a2.48 2.48 0 0 0 4.71 0H15V7h-.37zM5.22 8.51a1.52 1.52 0 0 1-.72.19 1.45 1.45 0 0 1-.71-.19A1.47 1.47 0 0 1 3.25 8h2.5a1.52 1.52 0 0 1-.53.51zM5.47 7h-2l1-2.4 1 2.4zm5.29 5L12 13.5H5L6.24 12h4.52zm1.78-7.38l1 2.4h-2l1-2.4zm.68 3.91a1.41 1.41 0 0 1-.72.19 1.35 1.35 0 0 1-.71-.19 1.55 1.55 0 0 1-.54-.53h2.5a1.37 1.37 0 0 1-.53.53z"
          />
        </svg>
        Ekosistem Stablecoin
      </button>
    </div>
  </div>
  <div
    class="lg:container pb-20"
  >
    <div
      class="overflow-x-auto"
    >
      <table
        cellpadding="0"
        cellspacing="0"
        class="table-auto border-separate loading-skeleton"
        data-testid="table-currency"
      >
        <thead>
          <tr>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  Crypto
                </div>
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                />
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  Harga
                </div>
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  24 Jam
                </div>
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  1 Mgg
                </div>
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  1 Bln
                </div>
              </button>
            </th>
            <th
              class="p-6 text-left border-gray border-b border-t first:pl-16 first:border-l lg:first:rounded-tl-xl last:border-r lg:last:rounded-tr-xl"
            >
              <button
                class="flex w-full text-left"
              >
                <div
                  class="font-bold uppercase flex justify-between w-full items-center text-gray-dark"
                >
                  1 Thn
                </div>
              </button>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
          <tr>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="400"
            >
              <div
                class="font-bold flex items-center gap-4"
              >
                <div
                  style="mask: url(?not-from-cache-please) no-repeat 50% 50%; background-image: url(?not-from-cache-please); background-size: cover; mask-size: cover; width: 32px; height: 32px; background-blend-mode: screen;"
                />
              </div>
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="200"
            >
              <div
                class="text-gray-dark font-normal"
              />
            </td>
            <td
              class=" p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="250"
            >
              <div
                class="font-bold "
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
            <td
              class="text-center p-6 border-gray border-b first:border-l last:border-r whitespace-nowrap"
              width="auto"
            >
              <div
                class="font-bold text-red"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
